packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/studio/services/representations/DomainDropTreeItemHandler.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.sirius.web.application.studio.services.representations;
+
+import java.util.Objects;
+
+import org.eclipse.sirius.components.collaborative.trees.api.IDropTreeItemHandler;
+import org.eclipse.sirius.components.collaborative.trees.dto.DropTreeItemInput;
+import org.eclipse.sirius.components.core.api.IEditingContext;
+import org.eclipse.sirius.components.representations.IStatus;
+import org.eclipse.sirius.components.trees.Tree;
+import org.eclipse.sirius.components.view.emf.IRepresentationDescriptionIdProvider;
+import org.eclipse.sirius.components.view.emf.IViewRepresentationDescriptionSearchService;
+import org.eclipse.sirius.web.application.views.explorer.services.ExplorerDescriptionProvider;
+import org.eclipse.sirius.web.application.views.explorer.services.ExplorerDropTreeItemHandler;
+import org.springframework.stereotype.Service;
+
+/**
+ * Provides the drop tree item tool for the Domain explorer by DSL explorer.
+ *
+ * @author gdaniel
+ */
+@Service
+public class DomainDropTreeItemHandler implements IDropTreeItemHandler {
+
+    private final ExplorerDropTreeItemHandler explorerDropTreeItemHandler;
```

Comment:
I somehow missed that you are using a dependency without an interface here. We are using interfaces on purpose, here we would need to extract the common behavior in a dedicated service instead. I'll update the PR and ensure that it gets merged.